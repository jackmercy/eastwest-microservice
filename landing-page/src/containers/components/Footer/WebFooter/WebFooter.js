import Link from 'next/link';
import React, { memo, useMemo } from 'react';
import { useIntl } from 'react-intl';

import {
  Register,
  CopyRight,
  SummaryInfo,
  SectionTitle,
  SwitchLanguages,
} from '../components';

import Content from '../../Content';
import useStyles from './webFooter.styles';

const WebFooter = () => {
  const intl = useIntl();
  const classes = useStyles();
  const configs = useMemo(
    () => [
      {
        key: 'product',
        title: intl.formatMessage({ id: 'FOOTER.PRODUCTS.TITLE' }),
        dataTest: 'footer-product',
        menu: [
          {
            key: 'talent-labs',
            href: '/talent-labs',
            label: intl.formatMessage({ id: 'FOOTER.PRODUCTS.TALENT_LABS' }),
          },
          // MVP hidden
          // {
          //   key: 'dashboard',
          //   href: '/dashboard',
          //   label: intl.formatMessage({ id: 'FOOTER.PRODUCTS.DASHBOARD' }),
          // },
          // {
          //   key: 'art',
          //   href: '/art',
          //   label: intl.formatMessage({ id: 'FOOTER.PRODUCTS.ART' }),
          // },
          // {
          //   key: 'admin',
          //   href: '/admin',
          //   label: intl.formatMessage({ id: 'FOOTER.PRODUCTS.ADMIN' }),
          // },
          // {
          //   key: 'rmt',
          //   href: '/rmt',
          //   label: intl.formatMessage({ id: 'FOOTER.PRODUCTS.RMT' }),
          // },
          {
            key: 'data-labs',
            href: '/data-labs',
            label: intl.formatMessage({ id: 'FOOTER.PRODUCTS.DATA_LABS' }),
          },
        ],
      },
      {
        key: 'company',
        title: intl.formatMessage({ id: 'FOOTER.COMPANY.TITLE' }),
        dataTest: 'footer-page',
        menu: [
          {
            key: 'about-us',
            href: '/company',
            label: intl.formatMessage({ id: 'FOOTER.COMPANY.ABOUT_US' }),
          },
          {
            key: 'careers',
            href: '/careers',
            label: intl.formatMessage({ id: 'FOOTER.COMPANY.CAREERS' }),
          },
          {
            key: 'contact',
            href: '/contact',
            label: intl.formatMessage({ id: 'FOOTER.COMPANY.CONTACT_US' }),
          },
        ],
      },
      {
        key: 'learn',
        title: intl.formatMessage({ id: 'FOOTER.LEARN.TITLE' }),
        dataTest: 'footer-other',
        menu: [
          {
            key: 'news',
            href: '/news-and-media',
            label: intl.formatMessage({ id: 'FOOTER.LEARN.NEWS' }),
          },
          // MVP hidden
          // {
          //   key: 'support',
          //   href: '/support',
          //   label: intl.formatMessage({ id: 'FOOTER.LEARN.SUPPORT' }),
          // },
          // {
          //   key: 'newsletter',
          //   href: '/newsletter',
          //   label: intl.formatMessage({ id: 'FOOTER.LEARN.NEWSLETTER' }),
          // },
        ],
      },
    ],
    [intl],
  );

  return (
    <footer className={classes.footer}>
      <Content>
        <div className={classes.content}>
          <SummaryInfo />
          {configs.map(({ key: parentKey, title, menu, dataTest }) => (
            <div key={parentKey} className={classes.group}>
              <SectionTitle title={title} className={classes.groupTitle} />
              {menu.map(({ key, label, href, ...restProps }) => (
                <Link key={`${parentKey}-${key}`} href={href} passHref>
                  <a
                    href="/#"
                    className={classes.groupItem}
                    data-test={dataTest}
                    {...restProps}
                  >
                    {label}
                  </a>
                </Link>
              ))}
            </div>
          ))}

          <div>
            <SwitchLanguages />
            <Register />
          </div>
        </div>
        <CopyRight />
      </Content>
    </footer>
  );
};

export default memo(WebFooter);
