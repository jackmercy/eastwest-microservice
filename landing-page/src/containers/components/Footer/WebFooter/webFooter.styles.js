import { createUseStyles } from 'react-jss';

import COLORS from '../../../../constants/COLORS';
import { hexToRgbA, rem } from '../../../../utils/jss';

export default createUseStyles({
  footer: { backgroundColor: COLORS.bgBlack01, overflow: 'hidden' },
  content: {
    display: 'flex',
    justifyContent: 'space-between',
    paddingTop: rem(112),
    paddingBottom: rem(88),
  },
  group: {},
  groupTitle: { marginBottom: rem(16) },
  groupItem: {
    display: 'block',
    textDecoration: 'none',
    fontSize: rem(16),
    lineHeight: rem(32),
    color: hexToRgbA(COLORS.textWhite, 0.5),
    '&:hover': { color: COLORS.primary },
  },
});
