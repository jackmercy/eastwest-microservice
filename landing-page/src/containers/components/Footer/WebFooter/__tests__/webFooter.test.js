import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import {
  Register,
  CopyRight,
  SummaryInfo,
  SwitchLanguages,
} from '../../components';
import WebFooter from '../WebFooter';

describe('Testing Footer > WebFooter', () => {
  let wrapper;

  beforeAll(() => {
    wrapper = mount(<WebFooter />);
  });

  it('should render match snapshot', () => {
    const wrapper = shallow(<WebFooter />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should have SummaryInfo', () => {
    expect(wrapper.find(SummaryInfo)).toHaveLength(1);
  });

  it('should have some product urls', () => {
    expect(wrapper.find('[data-test="footer-product"]')).toHaveLength(2);
  });

  it('should have some page urls', () => {
    expect(wrapper.find('[data-test="footer-page"]')).toHaveLength(3);
  });

  // MVP hidden
  // it('should have some other urls', () => {
  //   expect(wrapper.find('[data-test="footer-other"]')).toHaveLength(3);
  // });

  it('should have a SwitchLanguages', () => {
    expect(wrapper.find(SwitchLanguages)).toHaveLength(1);
  });

  it('should have a Register', () => {
    expect(wrapper.find(Register)).toHaveLength(1);
  });

  it('should have a CopyRight', () => {
    expect(wrapper.find(CopyRight)).toHaveLength(1);
  });
});
