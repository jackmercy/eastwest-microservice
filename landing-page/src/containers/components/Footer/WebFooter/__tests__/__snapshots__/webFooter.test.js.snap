// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Footer > WebFooter should render match snapshot 1`] = `
<footer
  className="footer-0-2-1"
>
  <Memo(Content)>
    <div
      className="content-0-2-2"
    >
      <Memo(SummaryInfo) />
      <div
        className="group-0-2-3"
        key="product"
      >
        <Memo(SectionTitle)
          className="groupTitle-0-2-4"
          title="Products"
        />
        <Link
          href="/talent-labs"
          key="product-talent-labs"
          passHref={true}
        >
          <a
            className="groupItem-0-2-5"
            data-test="footer-product"
            href="/#"
          >
            Talent labs
          </a>
        </Link>
        <Link
          href="/data-labs"
          key="product-data-labs"
          passHref={true}
        >
          <a
            className="groupItem-0-2-5"
            data-test="footer-product"
            href="/#"
          >
            Data labs
          </a>
        </Link>
      </div>
      <div
        className="group-0-2-3"
        key="company"
      >
        <Memo(SectionTitle)
          className="groupTitle-0-2-4"
          title="Company"
        />
        <Link
          href="/company"
          key="company-about-us"
          passHref={true}
        >
          <a
            className="groupItem-0-2-5"
            data-test="footer-page"
            href="/#"
          >
            About us
          </a>
        </Link>
        <Link
          href="/careers"
          key="company-careers"
          passHref={true}
        >
          <a
            className="groupItem-0-2-5"
            data-test="footer-page"
            href="/#"
          >
            Careers
          </a>
        </Link>
        <Link
          href="/contact"
          key="company-contact"
          passHref={true}
        >
          <a
            className="groupItem-0-2-5"
            data-test="footer-page"
            href="/#"
          >
            Contact us
          </a>
        </Link>
      </div>
      <div
        className="group-0-2-3"
        key="learn"
      >
        <Memo(SectionTitle)
          className="groupTitle-0-2-4"
          title="Learn"
        />
        <Link
          href="/news-and-media"
          key="learn-news"
          passHref={true}
        >
          <a
            className="groupItem-0-2-5"
            data-test="footer-other"
            href="/#"
          >
            News & Media
          </a>
        </Link>
      </div>
      <div>
        <Memo(SwitchLanguages) />
        <Memo(Register) />
      </div>
    </div>
    <Memo(CopyRight) />
  </Memo(Content)>
</footer>
`;
